clients/pubsub-common/src/main/java/org/eclipse/hono/client/pubsub/PubSubBasedAdminClientManager.java

```diff
@@ -171,37 +169,23 @@ private Future<Void> getOrCreateTopic(final TopicName topic) {
     }
 
     private Future<Void> getOrCreateTopic(final String projectId, final TopicName topic) {
-        if (topicExists(topic.toString())) {
+        if (topics.contains(topic.toString())) {
```

Comment:
IMHO we should move this check to the beginning of the *getOrCreateTopic(TopicName)* method in order to also prevent the (unnecessary) creation of the TopicAdminClient ...